files/libs/xroad_libs.sh

```diff
@@ -41,189 +41,184 @@ function destroy_api_key () {
     "https://localhost:4000/api/v1/api-keys/${api_key_id}"
 }
 
-function generate_csr () {
-  local type=$1
-  local id=$(readlink $type|cut -d . -f "1")
+function request_api () {
+  local verb=$1
+  local path=$2
+  local body=$3
 
-  # create csr if symlink to csr doesn't exist, set key type from $type first character
-  if [ ! -f ${id}.csr ]; then
-    log "generating certificate signing request for $type"
-    signer_console generate-cert-request ${id} \"$PX_INSTANCE $PX_MEMBER_CLASS $PX_MEMBER_CODE\" "${type:0:1}" "C=${PX_INSTANCE},O=${PX_MEMBER_CLASS},CN=${PX_MEMBER_CODE}" pem > /dev/null
-  else
-    log "certificate signing request for $type exists"
-  fi
+  local api_key=${created_api_key[1]}
+  local curl_args=('-H' 'Content-Type: application/json' '-H' "Authorization: X-Road-ApiKey token=$api_key")
+  api_response=$(curl -s -k -i -X $verb --data "$body" "${curl_args[@]}" "https://localhost:4000/api/v1$path")
+  api_response_status_code=$(echo -e "${api_response}"|head -n 1|cut -d$' ' -f2)
+  api_response_body=$(echo -e "${api_response}"|tail -n 1)
 }
 
-function generate_key () {
-  local type=$1
 
-  # create key if symlink to container doesn't exist
-  if [ ! -L ${type} ]; then
-    log "creating $type key"
-    signer_console generate-key 0 \"${type} key\" > /tmp/generate-key.log
-    # retrieve the keyId logged in stdout
-    id=$(cat /tmp/generate-key.log | grep "keyLabel" | jq -r .message | jq -r .data.keyId)
-    ln -s ${id}.p12 ${type}
-  else
-    log "$type key exists"
-  fi
-}
-
-function import_auth_certificate () {
-  local id=$(readlink $1|cut -d . -f "1")
+function initialize_security_server() {
+  local pin=$1
 
-  # import certificate if not already imported
-  if [ ! "$(signer_console list-certs | grep -A 1 ${id}|grep saved)" ]; then
-    log "importing auth certificate as saved state"
-    signer_console import-certificate-saved ${id}.crt \"$PX_INSTANCE $PX_MEMBER_CLASS $PX_MEMBER_CODE\" > /tmp/signer_console.log
-    # signer_console exit code is always 0 :(
-    if [ "$(cat /tmp/signer_console.log | grep ERROR)" ]; then
-      cat /tmp/signer_console.log
-      exit 1
+  request_api GET "/initialization/status"
+  if [ $api_response_status_code = 200 ]; then
+    software_token_init_status=$(echo $api_response_body | jq -r '.software_token_init_status')
+    if [ $software_token_init_status = NOT_INITIALIZED ]; then
+      software_token_pin=$(cat $autologin_file)
+      tpl='{
+  "owner_member_class": "%s",
+  "owner_member_code": "%s",
+  "security_server_code": "%s",
+  "software_token_pin": "%s",
+  "ignore_warnings": true
```

Comment:
this must be true, because otherwise it fails with warning `init_server_id_exists`